node/network/src/protocol.rs

```diff
@@ -241,12 +242,21 @@ impl Protocol {
     ) {
         info!("Send payload to {}", peer_id);
         if let Some(ch) = self.peer_manager.get_peer_channel(peer_id) {
-            let data = Encode::encode(&Message::PayloadResponse(request_id, payload)).unwrap();
-            forward_msg(ch, PeerMessage::Message(data));
+            let message = Message::PayloadResponse(request_id, payload);
+            self.send_message(ch, &message);
         } else {
             debug!(target: "network", "[SND PAYLOAD RSP] Channel for {} not found, account_id={:?}", peer_id, self.peer_manager.node_info.account_id);
         }
     }
+
+    /// Pass message through active proxies handlers and result messages are sent over channel `ch`.
+    /// Take owner of `message`.
```

Comment:
If it takes ownership, why message is a reference?

Reply:
Discussed in the #development channel.